### PR TITLE
Change the echo action to post an echo message to the connector API

### PR DESCRIPTION
**Pull request by kgrashad**
With this change you can actually chat with the elixir echo app from the Bot Framework Emulator (https://docs.botframework.com/en-us/tools/bot-framework-emulator/). Run the emulator and insert "http://localhost:4000/api/echo" in the Bot URL and start chatting.

### CODE REVIEW

**Comment by kgrashad** [comment]
@iassal  Let's make sure to squash when merging this pull-request to keep history clean.

File: echo_bot/mix.exs [hunk 1]
```diff
@@ -33,6 +33,7 @@ defmodule EchoBot.Mixfile do
      {:phoenix_html, "~> 2.6"},
      {:phoenix_live_reload, "~> 1.0", only: :dev},
      {:gettext, "~> 0.11"},
-     {:cowboy, "~> 1.0"}]
+     {:cowboy, "~> 1.0"},
+     {:botbuilder, path: "../../elixir-bot-builder", app: false}]
```
Comment (yshahin) [inline echo_bot/mix.exs:37]: Shouldn't this be the github path? Since it will be public
This relative path is for development

Comment (kgrashad) [reply echo_bot/mix.exs:37]: I thought we keep it this way for faster development until we have out first version ready for publishing.

**Comment by yshahin** [comment]
Cool

**Comment by kgrashad** [comment]
@yshahin Can you accept the PR?

File: echo_bot/web/controllers/echo_controller.ex [hunk 2]
```diff
@@ -1,18 +1,24 @@
 defmodule EchoBot.EchoController do
```
Comment (iassal) [inline echo_bot/web/controllers/echo_controller.ex:1]: update the tests so they won't fail. Run them by "mix test"

Comment (kgrashad) [reply echo_bot/web/controllers/echo_controller.ex:1]: Fixed

File: echo_bot/test/controllers/echo_controller_test.exs [hunk 3]
```diff
@@ -11,17 +11,39 @@ defmodule EchoBot.EchoControllerTest do
...
+        "id": "56800324",
+        "name": "Bot1"
+      },
+      "text": "hi",
+      "attachments": [],
+      "entities": []
+    }
     conn = post conn, echo_path(conn, :echo, params)
-    assert Map.equal?(json_response(conn, 200), %{"echoed_message" => "test message"})
+    assert Map.equal?(json_response(conn, 202), %{})
```
Comment (iassal) [inline echo_bot/test/controllers/echo_controller_test.exs:41]: Should you make sure that the text is hi? something like %{"text" => "hi"}

**Review by iassal — APPROVED** [review]

**Pull request merged**